phoenix-core/src/main/java/org/apache/phoenix/compile/ScanRanges.java

```diff
@@ -237,7 +237,9 @@ public void initializeScan(Scan scan) {
         return temp;
     }
     
-    public Scan intersectScan(Scan scan, final byte[] originalStartKey, final byte[] originalStopKey, final int keyOffset, boolean crossesRegionBoundary) {
+    public Scan intersectScan(Scan scan, final ImmutableBytesWritable originalStartKeyPtr, final ImmutableBytesWritable originalStopKeyPtr, final int keyOffset, boolean crossesRegionBoundary) {
+        byte[] originalStartKey= originalStartKeyPtr.get();
+        byte[] originalStopKey= originalStopKeyPtr.get();
```

Comment:
You can't treat an ImmutableBytesWritable the same as what was a byte[] before because an ImmutableBytesWritable has an offset and a length. By doing it this way, you're assuming that the offset is 0 and the length is byte[].length. Instead, you'd want to change the type and adjust the code as necessary:

    ImmutableBytesWritable originalStartKey= originalStartKeyPtr;
    ImmutableBytesWritable originalStopKey = originalStopKeyPtr;